src/Algebra/Structures.agda

```diff
@@ -567,3 +567,31 @@ record IsBooleanAlgebra
     ¬-cong                : Congruent₁ ¬
 
   open IsDistributiveLattice isDistributiveLattice public
+ 
+record IsUnitalMagma (∙ : Op₂ A) (ε : A) : Set (a ⊔ ℓ) where
+  field
+    isMagma : IsMagma ∙
+    identity : Identity ε ∙
+
+  open IsMagma isMagma public
+
+  identityˡ : LeftIdentity ε ∙
+  identityˡ = proj₁ identity
+
+  identityʳ : RightIdentity ε ∙
+  identityʳ = proj₂ identity
+
+record IsSemiGroupoid (∙ : Op₂ A) : Set (a ⊔ ℓ) where
+  field
+    assoc : Associative ∙
+
+record IsSmallCategory (∙ : Op₂ A) (ε : A) : Set (a ⊔ ℓ)  where
```

Comment:
Isn't this equivalent to `Monoid`, which is already defined?

Reply:
No, SmallCategory and SemiGroupoid need not satisfy "totality" as per my understanding.

Reply:
Right, but in Agda, all operations are total - you need to do extra work to do otherwise. So in Agda's internal logic, `IsSmallCategory` is equivalent to `Monoid`.

Reply:
Okay, then semiGroupoid is same as Semigroup?

Reply:
Yes, when done like that. SemiGroupoid is better seen as a unitless category, and as such would better fit in `agda-categories`.

Reply:
Okay. I will remove redundant definitions.